docs/conf.py

```diff
@@ -110,7 +110,7 @@
 autodoc_member_order = "bysource"
 
 # Default options for all autodoc directives
-autodoc_default_options = {
+autodoc_default_options: dict[str, Any] = {
```

Comment:
I don't see how these kind of type annotations are better

Reply:
Yeah, the thing is, pyright would otherwise mark this as `dict[str, Unknown]` and with `Unknown` being banned, if a wildcard type is indeed what we want, it must be declared explicitly like this. That's also one of the reasons why these rules might be a bit too strict.